src/commons/services/utils.js

```diff
@@ -108,6 +108,35 @@
             return clusterName;
         }
 
+        vm.getVolumeDetails = function(clusterId) {
+            volumeList = [];
+            clusterData = vm.clusterData.clusters;
+            len = clusterData.length;
+            for ( i = 0; i < len; i++ ) {
+                clusterObj = clusterData[i];
+                for( key in clusterObj ) {
+                    if(clusterId !== undefined) {
+                        if(key !== "stats" && key === clusterId) {
+                            if(clusterObj[key].volumes !== undefined) {
+                                for(index in clusterObj[key].volumes) {
+                                    volumeList.push(clusterObj[key].volumes[index])
```

Comment:
And also add a TODO for this as when API will support for array, the logic will change for this.